dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java

```diff
@@ -106,7 +106,10 @@ public FieldFilterService( FieldPathHelper fieldPathHelper, ObjectMapper jsonMap
         fieldPathHelper.apply( fieldPaths, HibernateProxyUtils.getRealClass( firstObject ) );
 
         SimpleFilterProvider filterProvider = getSimpleFilterProvider( fieldPaths, params.isSkipSharing() );
-        ObjectMapper objectMapper = jsonMapper.setFilterProvider( filterProvider );
+
+        // only set filter provider on a local copy so that we don't affect
+        // other object mappers (running across other threads)
+        ObjectMapper objectMapper = jsonMapper.copy().setFilterProvider( filterProvider );
```

Comment:
This is the only line that actually changes any behavior (the rest is cleanup + docs). This makes sure that a copy of the object mapper is created and so should be safe across threads/requests.